src/sys/unix.rs

```diff
@@ -1789,6 +1789,62 @@ impl crate::Socket {
         }
     }
 
+    /// Get the value for the `SO_ORIGINAL_DST` option on this socket.
+    ///
+    /// This value contains the original destination IPv4 address of the connection
+    /// redirected using `iptables` `REDIRECT` or `TPROXY`.
+    #[cfg(all(
+        feature = "all",
+        any(target_os = "android", target_os = "fuchsia", target_os = "linux")
```

Comment:
Note: I've added `fuchsia` based on presence of `libc::SO_ORIGINAL_DST` in Rust's libc for Fuchsia, but I have no way to verify that it actually works on that platform. Let me know if you want me to remove it.